MSWebdriver now fetches the binary corresponding to the currently installed Edge version instead of the latest one.

kapoorlakshya opened the pull request:
This PR fixes issue #14 and updates the README.

Since `MicrosoftWebDriver.exe` is not backwards compatible, we cannot always update to the latest version. This PR allows `webdrivers` gem to download the correct version for the currently installed version of Edge.

It also fixes the bug where the gem would re-download the binary when `current != released`. It now assumes that an existing binary is the correct version and skips the download process.

titusfortner commented:
Thanks for putting this together. Since we have subclasses for each driver, we don't want to add driver based conditionals to the superclass. What I'm thinking is that we can create `#correct_binary?` method and pull out:
```
current == released && File.exist?(binary)
```
Then `MSWebdriver#correct_binary?` would be:
```
File.exist?(binary)
```
Same with `download`. Subclass it for `MSWebdriver` and just change default:
```
def download(version = current)
  super
end
```

kapoorlakshya commented:
> Since we have subclasses for each driver, we don't want to add driver based conditionals to the superclass.

Good point, I definitely agree.  Will push the changes tonight.

kapoorlakshya commented:
@titusfortner I have updated the PR based on your suggestions. I have also fixed a failing `geckodriver` spec and all specs pass locally on Windows 10.

```
$ rspec
..........................

Finished in 29.82 seconds (files took 0.38402 seconds to load)
26 examples, 0 failures
```

However, the last `MSWebdriver` spec is failing on Travis because `current` relies on Powershell. Let me know if you want me to remove/modify this spec and what you think of the new changes.

kapoorlakshya committed:
MSWebdriver now fetches the binary corresponding to the currently installed Edge version instead of the latest one.

titusfortner commented:
Excellent, thanks! I just released 3.3.1
